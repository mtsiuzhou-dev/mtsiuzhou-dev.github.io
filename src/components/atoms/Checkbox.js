import PropTypes from 'prop-types';
import style from './Checkbox.module.scss';

const Checkbox = ({ id, className = '', isChecked = false, onChange = '' }) => (
  <input
    id={id}
    type="checkbox"
    className={`${style.checkbox} ${className || ''}`.trim()}
    defaultChecked={isChecked}
    onChange={onChange}
  />
);
Checkbox.defaultProps = {
  className: '',
  isChecked: false,
  onChange: null,
};
Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  isChecked: PropTypes.bool,
  onChange: PropTypes.func,
};
export default Checkbox;
