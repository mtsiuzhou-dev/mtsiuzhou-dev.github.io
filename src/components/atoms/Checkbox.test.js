import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

describe('Checkbox:', () => {
  const id = 'checkboxId';

  test('renders correctly', () => {
    const checkbox = renderer.create(<Checkbox id={id} />).toJSON();
    expect(checkbox).toMatchSnapshot();
  });

  test('onChange should be called', () => {
    const mockOnChange = jest.fn();
    render(<Checkbox id={id} onChange={mockOnChange} />);
    userEvent.click(screen.getByRole('checkbox'));

    expect(mockOnChange).toHaveBeenCalled();
  });
});
