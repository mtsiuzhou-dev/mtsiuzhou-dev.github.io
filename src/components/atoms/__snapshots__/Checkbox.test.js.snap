// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox: renders correctly 1`] = `
<input
  className="checkbox"
  defaultChecked={false}
  id="checkboxId"
  onChange={null}
  type="checkbox"
/>
`;
